halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java

```diff
@@ -166,30 +136,15 @@ public void saveConfig() {
       );
     }
 
-    Writer writer = null;
     try {
-      writer = new BufferedWriter(
-          new OutputStreamWriter(new FileOutputStream(halconfigPath), "utf-8"));
-      writer.write(yamlParser.dump(objectMapper.convertValue(halconfig, Map.class)));
-    } catch (Exception e) {
+      fileWriter.write(yamlParser.dump(objectMapper.convertValue(halconfig, Map.class)), halconfigPath);
```

Comment:
I think I'll replace the append operation with a string builder, and then write that to disk atomically instead.